### ISSUE TITLE
Facebook : Nominative address support

### ISSUE THREAD

**Issue by karlo2105** [streamlink/streamlink#880]
### Checklist

- [ ] This is a bug report.
- [x] This is a feature request.
- [ ] This is a plugin (improvement) request.
- [ ] I have read the contribution guidelines.

### Description

Facebook is a social video platform which supports live streaming. 

### Expected / Actual behavior

```
streamlink "https://www.facebook.com/IndexVijesti/videos/1906400442908386/" best
[cli][info] Found matching plugin facebook for URL https://www.facebook.com/Inde
xVijesti/videos/1906400442908386/
[cli][info] Available streams: 360p (best, worst)
[cli][info] Opening stream: 360p (hls)
[cli][info] Starting player: C:\Users\Ddr\Downloads\VLCPortable\App\vlc\vlc
[cli][info] Player closed
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

```
streamlink "https://www.facebook.com/IndexVijesti/"
error: No plugin can handle URL: https://www.facebook.com/IndexVijesti/

```
### Environment details

Operating system and version: Windows 7
Streamlink and Python version: Streamlink portable

**Comment by johnthecracker** [comment]
C:\>youtube-dl.exe -v -F --no-check-certificate https://www.facebook.com/IndexVijesti/videos/1892803307601433
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-F', '--no-check-certificate', 'https://www.facebook.com/IndexVijesti/videos/1892803307601433']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2017.05.01
[debug] Python version 3.4.4 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg n3.2.1, rtmpdump 2.4
[debug] Proxy map: {}
[facebook] 1892803307601433: Downloading webpage
[info] Available formats for 1892803307601433:
format code               extension  resolution note
247717965692890ad         m4a        audio only [eng] DASH audio   66k , mp4a.40.2 (48000Hz)
1419985091395935vd        mp4        428x240    [eng] DASH video  253k , avc1.4d401e, video only
122107244999658v          mp4        642x360    [eng] DASH video  503k , avc1.4d401e, video only
dash_sd_src               mp4        unknown
dash_sd_src_no_ratelimit  mp4        unknown
dash_hd_src               mp4        unknown
dash_hd_src_no_ratelimit  mp4        unknown    (best)


youtube-dl.exe -f 122107244999658v+247717965692890ad --no-check-certificate https://www.facebook.com/IndexVijesti/videos/1892803307601433



C:\MISC\YOUTUBE-DL>youtube-dl.exe -f 122107244999658v+247717965692890ad --no-check-certificate https://www.facebook.com/IndexVijesti/videos/1892803307601433
[facebook] 1892803307601433: Downloading webpage
[download] Destination: HOS slavi obljetnicu u Splitu-1892803307601433.f122107244999658v.mp4
[download]   6.9% of 123.79MiB at  1.18MiB/s ETA 01:37


Urlsnooper does not show url at all, I was not good at using urls from httpfox


[url=https://postimg.org/image/4qmn87ktr/][img]https://s27.postimg.org/4qmn87ktr/Untitled.png[/img][/url]

**Comment by johnthecracker** [comment]
C:\>youtube-dl.exe -v -F --no-check-certificate https://www.facebook.com/IndexVijesti/videos/1906400442908386/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-F', '--no-check-certificate', 'https://www.facebook.com/IndexVijesti/videos/1906400442908386/']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2017.05.01
[debug] Python version 3.4.4 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg n3.2.1, rtmpdump 2.4
[debug] Proxy map: {}
[facebook] 1906400442908386: Downloading webpage
[info] Available formats for 1906400442908386:
format code               extension  resolution note
1467887539929888ad        m4a        audio only DASH audio   66k , mp4a.40.2 (48000Hz)
1847991845521497vd        mp4        426x240    DASH video  208k , avc1.4d401e, video only
1445550678852431v         mp4        640x360    DASH video  407k , avc1.4d401e, video only
616764955178999v          mp4        854x480    DASH video  641k , avc1.4d401e, video only
613828672157429v          mp4        1280x720   DASH video 1368k , avc1.4d401e, video only
dash_sd_src               mp4        unknown
dash_sd_src_no_ratelimit  mp4        unknown
dash_hd_src               mp4        unknown
dash_hd_src_no_ratelimit  mp4        unknown    (best)



youtube-dl.exe -f 616764955178999v+1467887539929888ad --no-check-certificate https://www.facebook.com/IndexVijesti/videos/1906400442908386/



C:\MISC\YOUTUBE-DL>youtube-dl.exe -f 616764955178999v+1467887539929888ad --no-check-certificate https://www.facebook.com/IndexVijesti/videos/1906400442908386/
[facebook] 1906400442908386: Downloading webpage
[download] Destination: UZIVO Nastavlja se burna rasprava u Saboru! Petrov podnio ostavku!-1906400442908386.f616764955178999v.mp4
[download] 100% of 90.72MiB in 01:02
[download] Destination: UZIVO Nastavlja se burna rasprava u Saboru! Petrov podnio ostavku!-1906400442908386.f1467887539929888ad.m4a
[download] 100% of 9.42MiB in 00:07
[ffmpeg] Merging formats into "UZIVO Nastavlja se burna rasprava u Saboru! Petrov podnio ostavku!-1906400442908386.mp4"
Deleting original file UZIVO Nastavlja se burna rasprava u Saboru! Petrov podnio ostavku!-1906400442908386.f616764955178999v.mp4 (pass -k to keep)
Deleting original file UZIVO Nastavlja se burna rasprava u Saboru! Petrov podnio ostavku!-1906400442908386.f1467887539929888ad.m4a (pass -k to keep)


[url=https://postimg.org/image/ei86cog8p/][img]https://s4.postimg.org/6phikpa9p/Untitled.png[/img][/url][url=https://postimage.io/]free image host[/url][url=https://certificity.com]certificity.com[/url]

**Comment by johnthecracker** [comment]
C:\MISC\YOUTUBE-DL>youtube-dl.exe --no-check-certificate --get-url https://www.facebook.com/IndexVijesti/videos/1906400442908386/
https://scontent.flhr4-2.fna.fbcdn.net/v/t42.9040-29/10000000_613828692157427_2494494040326668288_n.mp4?efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfbGl2ZV9oZDFfZnJhZ18yX3ZpZGVvIn0=&oh=00163c836dd5c0fda810fd6e0165ecb6&oe=590B83A1
https://scontent.flhr4-2.fna.fbcdn.net/v/t42.1790-29/18272264_1467887553263220_7257170132261142528_n.mp4?efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfbGl2ZV9tZF9mcmFnXzJfYXVkaW8ifQ==&oh=a8f1852cf1d8c873534e85439a70639e&oe=590B82D3
